### PR TITLE
GSW-403 feat: route rpc

### PR DESCRIPTION
**Pull request by r3v4s**

### CODE REVIEW

File: pool/pool_router.gno [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	for poolPath, _ := range pools {
+		if strings.HasPrefix(poolPath, partialPath) {
+			foundPool = append(foundPool, poolPath)
+		}
+	}
+
+	// check if amount is enough
+	firstSelectedPool := []string{}
+	if false {
```
Comment (harryoh) [inline pool/pool_router.gno:35]: Is this for debugging?

File: pool/pool_router.gno [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		continue
+	} else if zeroForOne == true && amountSpecified > 0 {
```
Comment (harryoh) [inline pool/pool_router.gno:37]: is there no case amountSpecified is zero?

File: pool/pool_router.gno [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+			pool := GetPoolFromPoolKey(singlePool)
+			if pool.balances.token0 > amountSpecified/poolPrice { // must be bigger (can't be equal due to fee)
+				firstSelectedPool = append(firstSelectedPool, singlePool)
+			}
+		}
+
+	} else if zeroForOne == false && amountSpecified < 0 {
+		for _, singlePool := range foundPool {
+			pool := GetPoolFromPoolKey(singlePool)
+			if pool.balances.token0 > (-1 * amountSpecified) { // must be bigger (can't be equal due to fee)
```
Comment (harryoh) [inline pool/pool_router.gno:72]: fee can't be to zero?

File: pool/pool_router.gno (hunk 1 above)
Comment (r3v4s) [reply pool/pool_router.gno:35]: nah, it's one of filtering algorithm

File: pool/pool_router.gno (hunk 2 above)
Comment (r3v4s) [reply pool/pool_router.gno:37]: yes, will panic if amountSpecified is `0`

File: pool/pool_router.gno (hunk 3 above)
Comment (r3v4s) [reply pool/pool_router.gno:72]: it depends on swap situation, just wanted to include safe margin

**Review by harryoh — APPROVED** [review]

**Pull request merged**